modules/schedule/index.js

```diff
@@ -0,0 +1,396 @@
+  remind: function(r, parts, reply, command, from, to, text, raw) {
+            if(parts.length < 2)
+            {
+              reply("Usage: !remind ([target]) \"timeframe\" \"text\"");
+              reply("       Optional target defaults no one (channel wide).");
+              return;
+            }
+            
+            var target, schedule, command;
+
+            if(parts.length == 2)
+            {
+              schedule = parts[0];
+              command  = parts[1];
+            }
+            else
+            {
+              target   = parts[0];
+              schedule = parts[1];
+              command  = parts[2];
+            }
+
+            if(schedule.match(/in|at|after|tomorrow|next|from/i))
+            {
+              // use sugar to handle relative dates--- moment/later sucks at this
+              var s = Date.create(schedule);
+              if(!s.isValid())
+                return reply("Invalid relative date provided.");
+              if(!s.isFuture())
+                return reply("Cannot make reminders for past dates/times.");
+
+              // massage into a cron format
+              schedule = s.utc().format('{m} {H} {d} {M} *');
+            }
+
+            reply(newSchedule({
+              'blame': from,
+              'created': new moment(),
+              'schedule': schedule,
+              'command': command,
+              'channel': to,
+              'calls': 1,
+              'target': target
+            }));
+          },
+  schedule: {
+    _default: function(x,y,reply) {
+      reply("Usage: !schedule [<add>|<remove>|<list>|<help>] [arguments]");
+      reply("       Also see !schedule help <subcommand> for more details");
+    },
+    help: {
+      _default: null, // to be aliased later to ^
+      add: function(x,y,reply) {
+        reply("Usage: !schedule add \"timeframe\" \"command or text\"");
+        reply("       Timeframe syntax is here: http://bunkat.github.io/later/parsers.html#text");
+        reply("       If a valid command isn't specified, it is treated as text to print.");
+      },
+      remove: function(x,y,reply) {
+        reply("Usage: !schedule remove [digest]");
+        reply("       Removes the schedule specified by the digest.");
+      },
+      list: function(x,y,reply) {
+        reply("Usage: !schedule list [offset=0]");
+        reply("       Provides a list of schedules, in pages, from the specified offset of 0.");
+      },
+    },
+    add: function(r, parts, reply, command, from, to, text, raw) {
+      if(parts.length !== 2) return reply("add must have *exactly* two arguments");
+
+      reply(newSchedule({
+        'blame': from,
+        'created': new moment(),
+        'schedule': parts[0],
+        'command': parts[1],
+        'channel': to,
+        'calls': -1
+      }));
+    },
+    list: function(x, parts, reply, command, from, to) {
+      var offset = 0;
+      var count  = 5;
+      var ci     = count;
+      var oi     = 0;
```

Comment:
I don't get the point of having both oi and offset; it appears that oi is redundant?